src/Rewrite.hs

```diff
@@ -108,16 +122,47 @@ quotedUrlsET log rwArgs =
       . Utils.runLog log
     $ quotedUrls rwArgs
 
+--------------------------------------------------------------------------------
+-- Redirect homepage when moved.
+redirectedUrls :: MonadIO m => (Text -> m ()) -> Args -> ExceptT Text m (Maybe Text)
+redirectedUrls log (Args _ attrPth drvFile _) = do
+  lift $ log ""
+  homepage <- Nix.getHomepageET attrPth
+  response <- liftIO $ do
+    manager <- HTTP.newManager HTTP.defaultManagerSettings
+    request <- HTTP.parseRequest (T.unpack homepage)
+    HTTP.httpLbs request manager
+  let status = statusCode $ HTTP.responseStatus response
+  if status `elem` [301, 308]
+    then do
+      lift $ log "Redirecting URL"
+      let headers = HTTP.responseHeaders response
+          location = lookup "Location" headers
+      case location of
+        Nothing -> do
+          lift $ log "Server did not return a location"
+          return Nothing
+        Just (decodeUtf8 -> newHomepage) -> do
+          File.replaceIO homepage newHomepage drvFile
+          lift $ log "Replaced homepage"
+          return $ Just $
+            "Replaced homepage by "
+              <> newHomepage
+              <> " due http "
+              <> (T.pack . show) status
+    else do
+      lift $ log "URL not redirected"
+      return Nothing
+
 --------------------------------------------------------------------------------
 -- Rewrite Rust on rustPlatform.buildRustPackage
 -- This is basically `version` above, but with a second pass to also update the
 -- cargoSha256 vendor hash.
 rustCrateVersion :: MonadIO m => (Text -> m ()) -> Args -> ExceptT Text m (Maybe Text)
 rustCrateVersion log args@(Args _ attrPth drvFile drvContents) = do
-  lift $ log "[rustCrateVersion]"
   if not (T.isInfixOf "cargoSha256" drvContents)
     then do
-      lift $ log "[rustCrateVersion] No cargoSha256 found"
```

Comment:
Why removing all the brackets?  These were quite useful to tell when the updater was in a section, and if one rewriter crashes or hangs it's useful to tell how far along it got.  We added these for a reason!

Reply:
They aren't removed, they got systematically prepended to the logger at one level up.

Reply:
Ahhhh ok, that's nice! :sweat_smile: